### PR TITLE
feat: improve production debugging dx

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Resolve SSR stack trace paths

(outputs modified for readability. consola has a fancy logger that does some similar magic but only for display)

Old:

```
ReferenceError: crashme is not defined                                                                                                                                      16:07:17
    at asyncData (examples/hello-world/pages/index.vue:13:0)
    at promisify (examples/hello-world/.nuxt/utils.js:259:0)
    at server.js:1:<col>
    at Array.map (<anonymous>)
    at module.exports.__webpack_exports__.default (examples/hello-world/.nuxt/server.js:227:0)
```

New:

```
ReferenceError: crashme is not defined                                                                                                                                      
    at asyncData ([path]/examples/hello-world/pages/index.vue:13:0)
    at promisify ([path]/examples/hello-world/.nuxt/utils.js:259:0)
    at [path]/examples/hello-world/.nuxt/dist/server/server.js:2582:23
    at Array.map (<anonymous>)
    at module.exports.__webpack_exports__.default ([path]/examples/hello-world/.nuxt/server.js:227:0)
```

### Make chunk names easy to debug

Old:

```
├── client
│   ├── 421fcb3837a373bcc316.js
│   ├── 43f6a8fec12d37fd9731.js
│   ├── 923c20ce304abf61b7d0.js
│   ├── f4d7116a13cbd5bad38e.js
│   ├── fb4666c8451b74cab31e.js
│   └── LICENSES
└── server
    ├── 7250d0198e376567f352.js
    ├── ae78742f8bb4962f2610.js
    ├── client.manifest.json
    ├── index.spa.html
    ├── index.ssr.html
    ├── server.js
    └── server.manifest.json
```

New:

```
├── client
│   ├── app.43f6a8f.js
│   ├── commons.app.f4d7116.js
│   ├── LICENSES
│   ├── pages
│   │   ├── about.923c20c.js
│   │   └── index.421fcb3.js
│   └── runtime.e630eac.js
└── server
    ├── client.manifest.json
    ├── index.spa.html
    ├── index.ssr.html
    ├── pages
    │   ├── about.js
    │   ├── about.js.map
    │   ├── index.js
    │   └── index.js.map
    ├── server.js
    ├── server.js.map
    └── server.manifest.json
```

Also, server.js is no longer minimized by terser. This helps making build little faster and also having correct sourcemap lines.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by lihbr** [comment]
This is so cool, hope it'll land on time for 2.13.x 👏

File: packages/config/src/config/build.js [hunk 1]
```diff
@@ -14,12 +14,12 @@ export default () => ({
...
     // { isDev, isClient, isServer }
-    app: ({ isDev, isModern }) => isDev ? `${isModern ? 'modern-' : ''}[name].js` : '[contenthash].js',
-    chunk: ({ isDev, isModern }) => isDev ? `${isModern ? 'modern-' : ''}[name].js` : '[contenthash].js',
-    css: ({ isDev }) => isDev ? '[name].css' : '[contenthash].css',
-    img: ({ isDev }) => isDev ? '[path][name].[ext]' : 'img/[contenthash:7].[ext]',
-    font: ({ isDev }) => isDev ? '[path][name].[ext]' : 'fonts/[contenthash:7].[ext]',
-    video: ({ isDev }) => isDev ? '[path][name].[ext]' : 'videos/[contenthash:7].[ext]'
+    app: ({ isDev, isModern }) => isDev ? `${isModern ? 'modern-' : ''}[name].js` : '[name].[contenthash:7].js',
+    chunk: ({ isDev, isModern }) => isDev ? `${isModern ? 'modern-' : ''}[name].js` : '[name].[contenthash:7].js',
+    css: ({ isDev }) => isDev ? '[name].css' : '[name].[contenthash:7].css',
```
Comment (manniL) [inline packages/config/src/config/build.js:19]: IIRC we discussed using `name` for css, images etc. again but back then it was classified as breaking change.

File: packages/server/src/middleware/error.js [hunk 2]
```diff
@@ -112,17 +101,42 @@ const readSourceFactory = ({ srcDir, rootDir, buildDir }) => async function read
...
       }
-      return
+    }
+    return fileName
+  }
+
+  err.stack = (_err.stack || '')
+    .split('\n')
+    .map((l) => {
+      const m = l.match(/\(([^)]+)\)|([^\s]+\.[^\s]+):/)
```
Comment (manniL) [inline packages/server/src/middleware/error.js:117]: Could we improve variable naming here so the code is a bit more readable? 😋

File: packages/config/src/config/build.js (hunk 1 above)
Comment (pi0) [reply packages/config/src/config/build.js:19]: Actually it was mostly to not exposing anything from source. But the fact that already even component names can be reversed, it is just making debugging harder IMO. But have no objections using name or not for assets ☺️

Comment (manniL) [reply packages/config/src/config/build.js:19]: Not using it for js files is understandable for me, but img/css/fonts/videos/... always felt a bit wrong, especially bc image names can help in various places.

Comment (pi0) [reply packages/config/src/config/build.js:19]: Exactly. Also probably can help for CEO (searching images without alt)

I will keep change as is and comments unresolved if there is anything potentially breaking.

Comment (manniL) [reply packages/config/src/config/build.js:19]: Yup, SEO is one of the things where this comes in handy. Mentioned that in my previous talks mostly (+ how to change in the nuxt config).

Great 👍🏻

**Review by atinux — DISMISSED** [review]

**Comment by pimlie** [comment]
@pi0 Is there a measurable difference in parsing/loading server.js if not minimized? When you run a node server then that shouldn't matter much, but if you are using a lambda then that server.js could need to be parsed on almost every request.

**Comment by pi0** [comment]
@pimlie Agree however we are not currently officially supporting serverless target. Builders/Packers may utilize a minimizer to reduce size. I added `build._minifyServer` to make it easier until we improve support wdyt?

**Comment by pimlie** [comment]
As long as its still configurable I am happy, thanks ;)

**Comment by OrkhanAlikhanov** [comment]
@pi0 Is there any way to disable this? My routes are deep and it seems with this change file urls get even deeper  I think I am running into `SplitChunksPlugin` truncating filenames. 

Here is my file tree:
```
pages/
├── current-item
│   ├── _n.vue
│   └── index.vue
├── current-item.vue
├── dashboards
│   ├── _dashboardId
│   │   ├── items
│   │   │   ├── _itemId
│   │   │   │   ├── details.vue
│   │   │   │   └── youritems
│   │   │   │       └── _youritemId.vue
│   │   │   ├── _itemId.vue
│   │   │   └── index.vue
│   │   └── index.vue
│   └── _dashboardId.vue
......
```
Examples of long urls that nuxt generated but have been truncated:
```
vendors.pages/current-vendors.pages/current-item/[_]n.pages/dashboards/[_]dashboardId/items/_itemId/youritems/_youritemId..4d40e1ae.142e679.js

vendors.pages/dashboards/[_]dashboardId/items/_itemId/details.pages/dashboards/[_]dashboardId/items/.9ad5534e.e5bd853.js
```
Nuxt is unable to load last url in the example, when I try to access it it responds with `/* script not found */` message. If I however rename it from `.9ad5534e.e5bd853.js` to `a.9ad5534e.e5bd853.js` it is loadable.
I don't know why `vendor.pages` nest files that much deep. Maybe cyclic dependency thing?

What do you think?

**Comment by pi0** [comment]
Hi @OrkhanAlikhanov. It had to be fixed with #7603 but seems `[_]` is preserved. As a temporary fix you can use:

```js
build: {
  filenames: { 
    chunk: '[contenthash:7].js'
  }
}
```

**Comment by OrkhanAlikhanov** [comment]
Thanks for the quick response. Actually #7603 made the bug apparent. Before it, the filename had `_n.pages` now it is `[_]n.pages` making it even longer and getting truncated. Thanks for the temporary solution.

I still wonder if that much nestedness is a bug or a feature. The url contains `dashboards/[_]dashboardId/items/_itemId/details.pages/dashboards/[_]dashboardId/items/...` but the if you look at the file tree, the file is `dashboards/_dashboardId/items/details.vue`, why does nuxt nest `dashboards` folder inside `details.pages` folder? How should we interpret that for the easy debugging in production?

**Comment by pi0** [comment]
@OrkhanAlikhanov Indeed it is not nice a file name when [concatenation](https://webpack.js.org/plugins/module-concatenation-plugin/) happens. Using `[name]` was meant to make tracing chunks easier but we may need a way to simplify it for this case.

**Comment by gkatsanos** [comment]
I digged in the documention but dont see what the optimal setting for source mapping would be. I'm attaching a debugger to 9229 but I don't get any breakpoints to bind in neither VS debugger nor Chrome dev tools Node inspector ..

**Pull request merged**